File: App10/app/src/main/java/com/example/app10/MainActivity41.java
package com.example.appj4;

import android.os.Bundle;

import android.content.Intent;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;


public class MainActivity extends AppCompatActivity {

    EditText txtEdit1;
    EditText txtEdit2;
    EditText txtEdit3;

    Button Button1;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        txtEdit1 = findViewById(R.id.txtEdit1);

        txtEdit2 = findViewById(R.id.txtEdit2);

        txtEdit3 = findViewById(R.id.txtEdit3);

        Button1 = findViewById(R.id.Button1);


    }

    public void IrActivity2(View view) {
        String Nombre = txtEdit1.getText().toString();
        String Carrera = txtEdit2.getText().toString();
        String Grupo = txtEdit3.getText().toString();

        if(!Nombre.isEmpty() && !Carrera.isEmpty() && !Grupo.isEmpty()) {
            Intent pantalla2 = new Intent(this, MainActivity2.class);
            pantalla2.putExtra("Nombre", Nombre);
            pantalla2.putExtra("Carrera", Carrera);
            pantalla2.putExtra("Grupo", Grupo);
            startActivity(pantalla2);
        }
    }
}

File: App10K/app/src/main/java/com/example/app10k/MainActivity41.kt
package com.example.app10k

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity41 : AppCompatActivity() {
    var txtEdit1: EditText? = null
    var txtEdit2: EditText? = null
    var txtEdit3: EditText? = null

    var Button1: Button? = null

    var BTNR4: Button? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main41)
        ViewCompat.setOnApplyWindowInsetsListener(
            findViewById(R.id.main)
        ) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        txtEdit1 = findViewById(R.id.txtEdit1)

        txtEdit2 = findViewById(R.id.txtEdit2)

        txtEdit3 = findViewById(R.id.txtEdit3)

        Button1 = findViewById(R.id.Button1)

        BTNR4 = findViewById(R.id.BTNR4)

        BTNR4?.setOnClickListener(View.OnClickListener { v: View? -> regresarAOriginal() })
    }

    fun IrActivity2(view: View?) {
        val Nombre = txtEdit1!!.text.toString()
        val Carrera = txtEdit2!!.text.toString()
        val Grupo = txtEdit3!!.text.toString()

        if (!Nombre.isEmpty() && !Carrera.isEmpty() && !Grupo.isEmpty()) {
            val pantalla2 = Intent(
                this,
                MainActivity42::class.java
            )
            pantalla2.putExtra("Nombre", Nombre)
            pantalla2.putExtra("Carrera", Carrera)
            pantalla2.putExtra("Grupo", Grupo)
            startActivity(pantalla2)
        }
    }

    private fun regresarAOriginal() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }
}